internal/catalogmetadata/types.go

```diff
@@ -98,17 +98,17 @@ func (b *Bundle) loadRequiredPackages() error {
 		if err != nil {
 			return fmt.Errorf("error determining bundle required packages for bundle %q: %s", b.Name, err)
 		}
-		for _, requiredPackage := range requiredPackages {
-			semverRange, err := bsemver.ParseRange(requiredPackage.VersionRange)
+		for i := range requiredPackages {
+			semverRange, err := bsemver.ParseRange(requiredPackages[i].VersionRange)
 			if err != nil {
 				return fmt.Errorf(
 					"error parsing bundle required package semver range for bundle %q (required package %q): %s",
 					b.Name,
-					requiredPackage.PackageName,
+					requiredPackages[i].PackageName,
 					err,
 				)
 			}
-			requiredPackage.SemverRange = &semverRange
```

Comment:
So the issue is that we were setting `SemverRange` on a copy and not on the instance from `requiredPackages`.